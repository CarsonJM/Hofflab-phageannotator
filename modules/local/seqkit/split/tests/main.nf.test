nextflow_process {

    name "Test Process: SEQKIT_SPLIT"
    script "../main.nf"
    process "SEQKIT_SPLIT"


    test("fasta.gz") {

        when {
            params {
                seqkit_split_args   = "-p 3"
            }
            process {
                """
                input[0] = Channel.of(
                    [
                        [id: 'test'],
                        file("$projectDir/assets/test-datasets/fastas/MEGAHIT-test1.contigs.fa.gz", checkIfExists: true)
                    ]
                )
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("fasta.gz - stub") {

        options "-stub"

        when {
            params {
                seqkit_split_args   = "-p 3"
            }
            process {
                """
                input[0] = Channel.of(
                    [
                        [id: 'test'],
                        file("$projectDir/assets/test-datasets/fastas/MEGAHIT-test1.contigs.fa.gz", checkIfExists: true)
                    ]
                )
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
