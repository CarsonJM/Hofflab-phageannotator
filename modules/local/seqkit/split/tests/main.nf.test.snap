{
    "fasta.gz - stub": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "dir": [
                    
                ],
                "fastx": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-06-13T08:40:05.617719434"
    },
    "fasta.gz": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "dir": [
                    
                ],
                "fastx": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-06-13T08:39:53.16080694"
    }
}